api/src/main/java/com/velocitypowered/api/event/connection/PreLoginEvent.java

```diff
@@ -28,6 +30,14 @@ public String getUsername() {
         return username;
     }
 
+    public boolean isOnlineMode() {
+        return onlineMode;
+    }
+    
+    public void setOnlineMode(boolean onlineMode) {
```

Comment:
I oppose this approach since it goes against the very idea of resulted events. I would prefer this to be part of the result: that is, please use a new result class that represents "allowed", "allowed with fake game profile", "denied with reason".